libr/core/cmd_zign.c

```diff
@@ -174,84 +174,81 @@ static void addFcnZign(RCore *core, RAnalFunction *fcn, const char *name) {
+	RSignGraph graph;
+	graph.cc = -1;
+	graph.nbbs = -1;
+	graph.edges = -1;
+	graph.ebbs = -1;
+	graph.bbsum = 0;
+
+	char *ptr;
+	RListIter *iter;
+	r_list_foreach (args, iter, ptr) {
 		if (r_str_startswith (ptr, "cc=")) {
-			graph->cc = atoi (ptr + 3);
+			graph.cc = atoi (ptr + 3);
 		} else if (r_str_startswith (ptr, "nbbs=")) {
-			graph->nbbs = atoi (ptr + 5);
+			graph.nbbs = atoi (ptr + 5);
 		} else if (r_str_startswith (ptr, "edges=")) {
-			graph->edges = atoi (ptr + 6);
+			graph.edges = atoi (ptr + 6);
 		} else if (r_str_startswith (ptr, "ebbs=")) {
-			graph->ebbs = atoi (ptr + 5);
+			graph.ebbs = atoi (ptr + 5);
 		} else if (r_str_startswith (ptr, "bbsum=")) {
-			graph->bbsum = atoi (ptr + 6);
+			graph.bbsum = atoi (ptr + 6);
 		} else {
 			return false;
 		}
 	}
-	return true;
-}
-
-static bool addCommentZign(RCore *core, const char *name, const char *args0, int nargs) {
-	const char *comment = args0;
-	return r_sign_add_comment (core->anal, name, comment);
-}
-
-static bool addNameZign(RCore *core, const char *name, const char *args0, int nargs) {
-	const char *realname = r_str_word_get0 (args0, 0);
-	return r_sign_add_name (core->anal, name, realname);
-}
-
-static bool addGraphZign(RCore *core, const char *name, const char *args0, int nargs) {
-	RSignGraph graph = {0};
-	if (!parseGraphMetrics (args0, nargs, &graph)) {
-		eprintf ("error: invalid arguments\n");
-		return false;
-	}
 	return r_sign_add_graph (core->anal, name, graph);
 }
 
-static bool addHashZign(RCore *core, const char *name, int type, const char *args0, int nargs) {
-	if (!args0) {
+static bool addHashZign(RCore *core, const char *name, int type, RList *args) {
+	if (r_list_length (args) != 1) {
+		eprintf ("error: invalid syntax\n");
 		return false;
 	}
-	int len = strlen (args0);
+	const char *hash = (const char *)r_list_get_top (args);
+	int len = strlen (hash);
 	if (!len) {
 		return false;
 	}
-	return r_sign_add_hash (core->anal, name, type, args0, len);
+	return r_sign_add_hash (core->anal, name, type, hash, len);
 }
 
-static bool addBytesZign(RCore *core, const char *name, int type, const char *args0, int nargs) {
-	const char *hexbytes = NULL;
+static bool addBytesZign(RCore *core, const char *name, int type, RList *args) {
 	ut8 *mask = NULL, *bytes = NULL, *sep = NULL;
-	int size = 0, blen = 0;
+	int size = 0;
 	bool retval = true;
 
-	if (nargs != 1) {
+	if (r_list_length (args) != 1) {
 		eprintf ("error: invalid syntax\n");
-		retval = false;
-		goto out;
+		return false;
 	}
 
-	hexbytes = r_str_word_get0 (args0, 0);
-	if ((sep = (ut8*) strchr (hexbytes, ':'))) {
-		blen = sep - (ut8*) hexbytes;
-		if (!blen || (blen & 1) || strlen ((char*) ++sep) != blen) {
+	const char *hexbytes = (const char *)r_list_get_top (args);
+	if ((sep = (ut8 *)strchr (hexbytes, ':'))) {
+		int blen = sep - (ut8 *)hexbytes;
+		if (!blen || (blen & 1) || strlen ((char *)++sep) != blen) {
```

Comment:
preincrements in conditionals with multiple statements can be problematic,because it may end up not being executed. in this case it seems to be fine, but maybe better to have sep++ above